### ISSUE TITLE
Refine error message in ApolloServerInterceptor: Include OperationName

### ISSUE THREAD

**Issue by semaphore3000** [apollographql/apollo-android#3038]
ApolloServerInterceptor produces some ApolloNetworkExceptions like "Failed to execute http call" which are not very specific. 

```
callBack.onFailure(ApolloNetworkException("Failed to execute http call", e))
```

The logger in the same class gets more information, i.e.

```
logger.e(e, "Failed to execute http call for operation %s", request.operation.name().name())
```

So my feature request is just this: 

Couldn´t you please include the operation name also in the ApolloNetworkExceptions? 

This would make debugging much easier.

**Comment by martinbonnin** [comment]
Makes complete sense. Do you want to open a PR?

**Comment by semaphore3000** [comment]
Thank you for the quick reply. Sure, I can open one.

**Comment by martinbonnin** [comment]
Closed by #3039
